src/main/java/io/stargate/sgv2/jsonapi/service/resolver/matcher/TableFilterResolver.java

```diff
@@ -75,20 +90,44 @@ private static List<DBFilterBase> findDynamic(CaptureExpression captureExpressio
 
     // TODO: How do we know what the CmdT of the JsonLiteral<CmdT> from .value() is ?
     for (FilterOperation<?> filterOperation : captureExpression.filterOperations()) {
+      final Object rhsValue = filterOperation.operand().value();
       if (captureExpression.marker() == DYNAMIC_TEXT_GROUP) {
         filters.add(
             new TextTableFilter(
                 captureExpression.path(),
                 NativeTypeTableFilter.Operator.from(
                     (ValueComparisonOperator) filterOperation.operator()),
-                (String) filterOperation.operand().value()));
+                (String) rhsValue));
       } else if (captureExpression.marker() == DYNAMIC_NUMBER_GROUP) {
         filters.add(
             new NumberTableFilter(
                 captureExpression.path(),
                 NativeTypeTableFilter.Operator.from(
                     (ValueComparisonOperator) filterOperation.operator()),
-                (BigDecimal) filterOperation.operand().value()));
+                (BigDecimal) rhsValue));
+      } else if (captureExpression.marker() == DYNAMIC_DOCID_GROUP) {
+        Object actualValue = ((DocumentId) rhsValue).value();
+        if (actualValue instanceof String) {
+          filters.add(
+              new TextTableFilter(
+                  captureExpression.path(),
+                  NativeTypeTableFilter.Operator.from(
+                      (ValueComparisonOperator) filterOperation.operator()),
+                  (String) actualValue));
+        } else if (actualValue instanceof BigDecimal) {
+          filters.add(
+              new NumberTableFilter(
+                  captureExpression.path(),
+                  NativeTypeTableFilter.Operator.from(
+                      (ValueComparisonOperator) filterOperation.operator()),
+                  (BigDecimal) actualValue));
+        } else {
+          throw new UnsupportedOperationException(
+              "Unsupported DocumentId type: " + rhsValue.getClass().getName());
+        }
+      } else {
+        throw new UnsupportedOperationException(
+            "Unsupported dynamic filter type: " + filterOperation);
```

Comment:
I know this is just following the pattern. We can change these to Data API exception later.

Reply:
Yes, sorry I missed this one before merging.